application/src/main/java/org/togetherjava/tjbot/commands/help/HelpThreadCommand.java

```diff
@@ -195,9 +197,24 @@ private void closeThread(SlashCommandInteractionEvent event, ThreadChannel helpT
             .setColor(HelpSystemHelper.AMBIENT_COLOR)
             .build();
 
+        manuallyResetChannelActivityCache.invalidate(helpThread);
         event.replyEmbeds(embed).flatMap(any -> helpThread.getManager().setArchived(true)).queue();
     }
 
+    private void resetActivity(SlashCommandInteractionEvent event, ThreadChannel helpThread) {
+        refreshCooldownFor(Subcommand.RESET_ACTIVITY, helpThread);
+
+        if (!event.getMember().equals(helpThread.getOwner())) {
+            event.reply("Only the thread creator can reset activities.").queue();
+        }
+
+        RestAction<List<Message>> messages = helpThread.getHistory().retrievePast(1);
+
+        manuallyResetChannelActivityCache.put(helpThread, messages.complete().get(0).getId());
```

Comment:
u can't use .complete method, it blocks the current thread. And we can't take that

Reply:
changed blocking to async.